core/src/main/java/com/datastax/oss/driver/internal/core/util/concurrent/CompletableFutures.java

```diff
@@ -120,4 +121,22 @@ public static Throwable getFailed(CompletableFuture<?> future) {
       }
     }
   }
+
+  /**
+   * Executes a function on the calling thread and returns result in a {@link CompletableFuture}.
+   *
+   * <p>Similar to {@link CompletableFuture#completedFuture} except takes a {@link Supplier} and if
+   * the supplier throws an unchecked exception, the returning future fails with that exception.
+   *
+   * @param supplier Function to execute
+   * @param <T> Type of result
+   * @return result of function wrapped in future
+   */
+  public static <T> CompletableFuture<T> completedFuture(Supplier<T> supplier) {
```

Comment:
Alternatively could have used `CompletableFuture.supplyAsync` with `MoreExectuors.directExecutor` but this seems more simple.